### DISCUSSION TITLE
tp.file.title plus header or section info?

### DISCUSSION THREAD

**Discussion by cedardocster** [SilentVoid13/Templater#804] [Help]
I have a template to make a link to a new note from the current note that works fine:

[[<% tp.file.title %>.note|note]]

So if I'm in the note called "Bob" it will insert the new link like this: [[Bob.note|note]]

What I'd like to do is if the most recent header above the note is:
# Some Section Header

the template would add in that header information like this: [[Bob.Some Section Header.note|note]]

That way I can have several unique note names across one note as long as they are all in different sections (have unique headers above them)

---

alternatively, is there a way to have some sort of force auto counting if there's already a note named that already in existence?  So with the first example you'd get 

[[Bob.note|note]]
[[Bob.note(1)|note]]
[[Bob.note(2)|note]]

Thanks!
- Dave

**Comment by welpdx** [comment]
Can you provide your template? I don't understand what is the issue here exactly because can't you just do what you did:
[[Bob.note|note]]
[[Bob.note(1)|note]]
[[Bob.note(2)|note]]

manually? unless you want to generate like 9 in a row and you don't want to copy paste it and add the numbers in there...

Thanks

**Reply by cedardocster** [reply]
the template is this:
[[<% tp.file.title %>.note|note]]

and when I use it any number of times it creates the exact same link:

[[Bob.note|note]]
[[Bob.note|note]]
[[Bob.note|note]]

so you can make multiple links, but they all go to the same note.

(I am super lazy - I just want the template to detect either the heading or the presence of the note and automatically fill in the right thing - that's what templates are for! :) )

**Reply by AB1908** [reply]
But isn't the link invalid? It doesn't follow the obsidian link syntax? Does it work if you do it by hand?

**Reply by cedardocster** [reply]
it works for me :)

it looks like "note" with a link format but goes to "Bob.note"
